eng/pipelines/libraries/build-job.yml

```
@@ -141,6 +141,7 @@ jobs:
             - template: /eng/pipelines/common/upload-intermediate-artifacts-step.yml
               parameters:
                 name: Libraries_AllConfigurations
+                condition: eq(variables['_librariesBuildProducedPackages'], true)
```

```suggestion
                condition: >-
                  or(
                     eq(variables['_librariesBuildProducedPackages'], true),
                     eq(variables['Build.SourceBranchName'], 'main'),
                     eq(variables['System.PullRequest.TargetBranch'], 'main')) 
```

This way we ensure that if there are no produced packages on main we fail.